spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2931,8 +2931,8 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
           msg = s"unsupported Spark partitioning expressions: $expressions"
         }
         supported
-      case SinglePartition => true
-      case RoundRobinPartitioning(_) => true
+      case SinglePartition => inputs.forall(attr => supportedDataType(attr.dataType))
+      case RoundRobinPartitioning(_) => inputs.forall(attr => supportedDataType(attr.dataType))
```

Comment:
I brought this regression in #760. We still need to check input data types for SinglePartition, etc.